### ISSUE TITLE
Error when attempting to define an `isolated` anonymous function in-line as an argument 

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#25780]
**Description:**
$title.

**Steps to reproduce:**
```ballerina
public function main() {
    foo(isolated function () {

    });
}

function foo(isolated function() x) {
    
}
```

```cmd
$ ballerina run test.bal 
Compiling source
        test.bal
error: .::test.bal:2:9: missing ellipsis token
error: .::test.bal:2:9: incompatible types: expected '[isolated function () returns ()]', found 'isolated function () returns ()'
```

**Affected Versions:**
`isolated-functions` branch with changes in https://github.com/ballerina-platform/ballerina-lang/pull/25763 and compile time analysis.

**Comment by MaryamZi** [comment]
Need to enable the following when fixing this
- https://github.com/ballerina-platform/ballerina-lang/pull/25782/commits/0f985a0c87f00c3a30dc51050899bb774d02bfc5#diff-a2ca38034aadd86d6a57b19f5dabb9dfR47
- https://github.com/ballerina-platform/ballerina-lang/pull/25782/commits/7c30bfe67e550f569fea4d1d58c0a6dc99e646e7

**Comment by rdulmina** [comment]
Fixed with #25808

**Comment by MaryamZi** [comment]
> Need to enable the following when fixing this
> 
> * [0f985a0#diff-a2ca38034aadd86d6a57b19f5dabb9dfR47](https://github.com/ballerina-platform/ballerina-lang/commit/0f985a0c87f00c3a30dc51050899bb774d02bfc5#diff-a2ca38034aadd86d6a57b19f5dabb9dfR47)
> * [7c30bfe](https://github.com/ballerina-platform/ballerina-lang/commit/7c30bfe67e550f569fea4d1d58c0a6dc99e646e7)

These were also enabled in https://github.com/ballerina-platform/ballerina-lang/pull/25761.
